frame/staking/src/pallet/impls.rs

```diff
@@ -17,7 +17,10 @@
 
```

Comment:
in this file it seems there is a miss in ElectionDataProvider::clear implementation: it doesn't remove CounterForValidator CounterForNominator, it doesn't reset the bags also.